src/fhir.js

```diff
@@ -301,15 +391,74 @@ class Patient extends FHIRObject {
       return [];
     }
     const resourceType = classInfo.name.replace(/^FHIR\./, '');
-    const records = this._bundle.entry.filter( e => {
-      return e.resource && e.resource.resourceType == resourceType;
-    }).map( e => {
-      return new FHIRObject(e.resource, classInfo, this._modelInfo);
-    });
+    const records = this._bundle.entry
+      .filter(e => {
+        return e.resource && e.resource.resourceType == resourceType;
+      })
+      .map(e => {
+        return new FHIRObject(e.resource, classInfo, this._modelInfo);
+      });
     return records;
   }
 }
 
+class AsyncPatient extends FHIRObject {
+  constructor(patientData, modelInfo, serverData) {
+    const patientClass = modelInfo.patientClassIdentifier
+      ? modelInfo.patientClassIdentifier
+      : modelInfo.patientClassName;
+    const ptClass = modelInfo.findClass(patientClass);
+    super(patientData, ptClass, modelInfo);
+
+    // Define a "private" un-enumerable property to hold the patient data
+    Object.defineProperty(this, '_patientData', {
+      value: patientData,
+      enumerable: false
+    });
+    Object.defineProperty(this, '_serverData', {
+      value: serverData,
+      enumerable: false
+    });
+  }
+
+  async findRecords(profile) {
+    const classInfo = this._modelInfo.findClass(profile);
+    if (classInfo == null) {
+      console.error(`Failed to find type info for ${profile}`);
+      return [];
+    }
+    const resourceType = classInfo.name.replace(/^FHIR\./, '');
+    // If the patient resource type is requested, return array with just this resource
+    if (resourceType === 'Patient') {
+      return [this];
+    }
+
+    const compartmentInfo = patientCompartmentDefinition.resource.filter(
+      def => def.code === resourceType
+    );
+    if (!compartmentInfo[0]) {
+      throw new Error(`Resource type: ${resourceType} cannot reference a patient.`);
+    }
```

Comment:
Hmmmmmm.... I feel like there could _maybe_ be cases where findRecords is called, still in a Patient context, where the resource queried for can't reference a patient.

Take EXM111 for example. Embedded within one of the CQL definitions, there's a [clause](https://github.com/DBCG/connectathon/blob/28d976d5e2ecf8ffa9badb82964395877323b63f/fhir401/bundles/measure/EXM111-9.1.000/EXM111-9.1.000-files/EXM111-9.1.000.cql#L72) in a query that calls `Global.GetLocation`. That function does a [retrieve](https://github.com/DBCG/connectathon/blob/28d976d5e2ecf8ffa9badb82964395877323b63f/fhir401/input/pagecontent/cql/MATGlobalCommonFunctions-5.0.000.cql#L230) for `[Location]`

As we see in the compartment def, `Location` can't reference a Patient, but the measure still wants to query for them. So rather than throwing an error here, the default behavior for resourceTypes that don't have a compartment def entry should probably be to just send a serach request to `/${resourceType}` with no additional query params and return all the resources that come back